port global chanops and logging of kicks from sigmabot

MacFan4000 opened the pull request.

MacFan4000 committed:
port global chanops and logging of kicks from sigmabot

RhinosF1 commented on sopel_channelmgnt/channelmgnt/__init__.py:
```diff
@@ -35,6 +42,9 @@ def configure(config):
     config.define_section('channelmgnt', ChannelmgntSection, validate=False)
     config.channelmgnt.configure_setting('datafile', 'Where is the datafile for channelmgnt?')
     config.channelmgnt.configure_setting('support_channel', 'What channel should users ask for help in?')
+    config.channelmgnt.configure_setting('global_chanops', 'If anybody, who should have chanop access in all channels? (if any then enter nickserv accounts)')
```
We support global_chanops now, it's called inherits from

MacFan4000 replied:
Does having chanops and inherits from defined at the same time work?

RhinosF1 replied:
see my other PR

MacFan4000 replied:
Ok so I can't add that for us until that is merged

RhinosF1 commented on sopel_channelmgnt/channelmgnt/__init__.py:
```diff
@@ -35,6 +42,9 @@ def configure(config):
     config.channelmgnt.configure_setting('datafile', 'Where is the datafile for channelmgnt?')
     config.channelmgnt.configure_setting('support_channel', 'What channel should users ask for help in?')
+    config.channelmgnt.configure_setting('global_chanops', 'If anybody, who should have chanop access in all channels? (if any then enter nickserv accounts)')
+    config.channelmgnt.configure_setting('log_kicks', 'Should kicks be logged to a channel? (true/false)')
```
use channel specific config formats in datafile

MacFan4000 replied:
?

RhinosF1 replied:
log_kicks should be per channel overrideable

MacFan4000 replied:
I was trying to mimic sigmabot, where kicks for all channels went to the same channel

RhinosF1 replied:
I'm saying that's not ideal. We should allow channels to have their own logs too.

RhinosF1 commented on sopel_channelmgnt/channelmgnt/__init__.py:
```diff
@@ -35,6 +42,9 @@ def configure(config):
     config.channelmgnt.configure_setting('support_channel', 'What channel should users ask for help in?')
+    config.channelmgnt.configure_setting('global_chanops', 'If anybody, who should have chanop access in all channels? (if any then enter nickserv accounts)')
+    config.channelmgnt.configure_setting('log_kicks', 'Should kicks be logged to a channel? (true/false)')
+    config.channelmgnt.configure_setting('log_channel', 'If you said true to the previous question, then please provide the channel where kicks should be logged to')
```
as above

MacFan4000 replied:
?

RhinosF1 reviewed (changes requested):
see review

MacFan4000 committed:
fix errors

MacFan4000 committed:
Update codeql-analysis.yml

MacFan4000 committed:
Update workflow.yml

MacFan4000 committed:
Update codeql-analysis.yml

MacFan4000 committed:
Update workflow.yml

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

RhinosF1 reviewed (changes requested):
Blocking on the PR to update channelmgnt get chanops logic. I don't want to make the issue worse.

MacFan4000 committed:
Merge branch 'dev' into MacFan4000-patch-1

MacFan4000 committed:
fix ci

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
add ability to have default set of chanops defined in json

"default" would be defined like any other channel, if it is defined then it is added to the chanops list, if it isn't then its not added

MacFan4000 committed:
fix

MacFan4000 committed:
Merge branch 'dev' into MacFan4000-patch-1

deepsource-autofix[bot] committed:
Autofix issues in 1 file

Resolved issues in sopel_channelmgnt/channelmgnt/__init__.py via DeepSource Autofix

MacFan4000 committed:
add logic for having a default logging channel

MacFan4000 committed:
fix

MacFan4000 commented:
> Blocking on the PR to update channelmgnt get chanops logic. I don't want to make the issue worse.

I honestly don't see how this would make it any worse.

RhinosF1 commented:
This PR abuse the same logic

RhinosF1 commented on sopel_channelmgnt/channelmgnt/__init__.py:
```diff
@@ -43,6 +43,10 @@ def configure(config):
     config.channelmgnt.configure_setting('support_channel', 'What channel should users ask for help in?')
 
 
+BOLD = '\x02'
```
use sopel's built in

RhinosF1 commented on sopel_channelmgnt/channelmgnt/__init__.py:
```diff
@@ -43,6 +43,10 @@ def configure(config):
 
 
+BOLD = '\x02'
+GREEN = '\x0303'
```
as with BOLD

RhinosF1 reviewed (changes requested):
https://sopel.chat/docs/api.html#module-sopel.formatting

MacFan4000 committed:
Merge branch 'dev' into MacFan4000-patch-1

MacFan4000 committed:
fixes

MacFan4000 committed:
more fixes

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update .flake8

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
Update __init__.py

MacFan4000 committed:
try to fix

MacFan4000 committed:
Update __init__.py

RhinosF1 reviewed (commented):
Do we have to increase line lengths or can lines be appropriately broke up?

MacFan4000 committed:
Update .flake8

RhinosF1 reviewed (approved):
Works for now, I'll build the new system on top later.
